### ISSUE TITLE
Add checkbox to toggle display of content blocks

### ISSUE THREAD

**Issue by ggeisler** [projecthydra-labs/hyku#813]
### Descriptive summary

Currently, in order to hide a content block from displaying, the administrator has to remove all content from the TinyMCE edit pane. If we instead made the content toggleable, it would make it easier for the admin to hide the block and enable the admin to reuse a block without having to input it again.

### Expected behavior
As an admin, I should be able to temporarily hide a content block without having to re-enter the block text and formatting.

### Actual behavior
Currently, to hide a content block I have to delete all the content from the block.

### Steps to reproduce the behavior

1. Go to Administration
1. Go to Configuration > Settings > Content Blocks
1. Enter text for any block and save it
1. Go back and try to hide your text from displaying without having to delete it

### Mockup

The mockup below shows the addition of a checkbox for toggling the display of the Featured Researcher content block. Ideally it would work like this:

* If there is no content in the TinyMCE input field, the checkbox should be in the off state (to prevent the block from displaying when there is no content to display)
* When there is content, the block is displayed when the checkbox is checked.

![content-block-toggle](https://cloud.githubusercontent.com/assets/101482/23189807/f0621be6-f851-11e6-9a1b-f5786c5e489f.png)

The other content blocks would work the same way:

- [ ] About page

- [ ] Announcement text

- [ ] Marketing text

But note that the About page is special in that it also has a navigation menu item. If the "Display About page" checkbox is unchecked, we should not show the About page in the navigaation menu.

**Comment by jcoyne** [comment]
If we don't display it, this would leave a rather unseemly white space in the place this is usually displayed.  Are you sure?

**Comment by ggeisler** [comment]
How is it different from displaying a block with no content?

**Comment by jcoyne** [comment]
I'm saying that both are bad behavior (with the exception of the "Announcement Text").  It is simpler if there is only one control.   And it is clearer to the user that they ought  to have something if there's no "disable". 

I'm not sure why I'd want to turn off the marketing text or especially the about page.

**Comment by ggeisler** [comment]
It sounds like you are assuming that every Hyku adopter is going to be content to have their homepage look the same, with a Featured Researcher block and a Marketing Text block. Especially as we're targeting a broader audience than IRs, my assumption is some adopters will not necessarily want to have a Featured Researcher or Marketing Text block.

By not giving the admin control to hide the display of a block or the About page, you're forcing them to put content in them even if they don't want to. (And in the case of the Announcement Text block, re-enter text every time they want to display an announcement, even if it is the same as their last announcement.)   It certainly makes sense that most sites would want an About page, but it seems presumptuous of us to force someone to have an About page. Again, as we're targeting a broader audience than just IRs from institutions we're familiar with, not forcing every adopter to follow the same pattern seems like a good thing.

**Comment by hannahfrost** [comment]
@jcoyne @ggeisler I agree that a toggle for display is highly desirable. Especially for Featured Researcher and Announcement Text, so we might as well do it for all the content blocks, as I expect that folks would make use of them too. 

I'm adding this to the Ready column, but down the list of priorities.

**Comment by mjgiarlo** [comment]
@jcoyne thinks this may be low priority except for announcement text. The others, if turned off, make the site look "awful." (Discussed on kick-off today.) Comments, @hannahfrost @ggeisler?

**Comment by ggeisler** [comment]
@mjgiarlo I won't argue the low priority aspect. I think the user should have control over displaying the blocks at some point. Maybe we wait until we improve the design of the homepage and there are more blocks to choose from, because in the long run forcing a user to show a Featured Researcher block on their homepage is not compatible with how @hannahfrost and I at least envision future adopters to wanting to use this product. That feels like a very backward-looking requirement.

**Comment by mjgiarlo** [comment]
@ggeisler Agree with that :100:.

**Comment by hannahfrost** [comment]
@mjgiarlo @ggeisler I can live with that.
